### PR TITLE
TW-160: Handle permission for Camera

### PR DESCRIPTION
**Pull request by nqhhdev**
- Issue: #160 
- Resolved:

`IOS`

https://github.com/linagora/twake-on-matrix/assets/99852347/da214156-e54a-4b0f-9df9-56ff2f6c78fc

https://github.com/linagora/twake-on-matrix/assets/99852347/21812cf2-060b-44f1-bfe5-872cbc0825c4

`Android`

https://github.com/linagora/twake-on-matrix/assets/99852347/4b89819d-ae36-40b5-9851-98688cd5eddb

https://github.com/linagora/twake-on-matrix/assets/99852347/b156909e-64dd-476f-b296-9162b095e3ee

### CODE REVIEW

File: lib/domain/usecase/send_image_interactor.dart [hunk 1]
```diff
@@ -1,20 +1,17 @@
...
     String? txId,
     Event? inReplyTo,
     String? editEventId,
     int? shrinkImageMaxDimension,
     Map<String, dynamic>? extraContent,
   }) async {
-    final matrixFile = await entity.toMatrixFile();
     if (matrixFile != null) {
       try {
         final mxcUri = await room.sendImageFileEvent(
```
Comment (sherlockvn) [inline lib/domain/usecase/send_image_interactor.dart:17]: use `sendImagesFileEvent` instead

File: lib/pages/chat/chat.dart [hunk 2]
```diff
@@ -399,11 +400,12 @@ class ChatController extends State<Chat> {
...
 
-  void sendImage() {
+  void sendImage() async {
     final assetEntity = imagePickerController.selectedAssets.first;
     final sendImageInteractor = getIt.get<SendImageInteractor>();
     if (assetEntity.asset.type == AssetType.image) {
-      sendImageInteractor.execute(room: room!, entity: assetEntity.asset);
+      final matrixFile = await assetEntity.asset.toMatrixFile();
+      sendImageInteractor.execute(room: room!, matrixFile: matrixFile);
       imagePickerController.clearAssetCounter();
```
Comment (sherlockvn) [inline lib/pages/chat/chat.dart:409]: can you refactor it using `removeAllImageSelected` like you do, please

File: lib/pages/chat/chat.dart [hunk 3]
```diff
@@ -1168,6 +1170,31 @@ class ChatController extends State<Chat> {
...
+    );
+
+    if (result == true) {
+      Navigator.pop(context);
+    }
+  }
+  void imagePickAction() async {
+    Navigator.pop(context);
+    final file = await ImagePicker().pickImage(source: ImageSource.camera);
+    if (file == null) return;
```
[the remark this answers is not in the document]
Comment (nqhhdev) [reply lib/pages/chat/chat.dart:1191]: Only null or not null. There is no empty case

[the remark this answers is not in the document]
Comment (sherlockvn) [reply lib/pages/chat/chat.dart:1191]: But there is case file is not null, but file is empty

File: lib/domain/usecase/send_image_interactor.dart [hunk 4]
```diff
@@ -1,20 +1,17 @@
...
 import 'package:fluffychat/presentation/extensions/room_extension.dart';
 import 'package:matrix/matrix.dart';
-import 'package:photo_manager/photo_manager.dart';
-import 'package:fluffychat/presentation/extensions/asset_entity_extension.dart';
 
 class SendImageInteractor {
   Future<void> execute({
     required Room room,
-    required AssetEntity entity,
+    MatrixFile? matrixFile,
```
Comment (hoangdat) [inline lib/domain/usecase/send_image_interactor.dart:8]: Why you do this, we expect to move the heavy task to `asynchronize` function. This is a kind of heavy task

**Review by hoangdat — APPROVED** [review]

**Pull request merged**